packages/template-retail-react-app/app/utils/utils.js

```diff
@@ -199,3 +199,20 @@ export const mergeMatchedItems = (arr1 = [], arr2 = []) => {
  * @return {boolean}
  */
 export const isHydrated = () => typeof window !== 'undefined' && !window.__HYDRATING__
+
+/**
+ * Find the smallest value by key from a given array
+ * @param arr
+ * @param key
+ */
+export const getSmallestValByProperty = (arr, key) => {
+    if (!arr || !arr.length) return undefined
+    if (!key) {
+        throw new Error('Please specify a key.')
+    }
+    const vals = arr
+        .map((item) => item[key])
+        .filter(Boolean)
+        .filter(Number)
+    return vals.length ? Math.min(...vals) : undefined
+}
```

Comment:
I think it might be a better idea to inline this as a private function in your product utils. I don't think it's usefulness is going to be that high and we'd but have to support it forever as it's a public api

Reply:
Sounds good, I have moved the function into product-utils.js